modules/lkqdBidAdapter.js

```diff
@@ -101,29 +61,101 @@ function buildRequests(validBidRequests) {
-      sspUrl = _replaceMacro(DNT_KEY, bidRequest.params.dnt, DNT_DEFAULT, sspUrl);
-      sspUrl = _replaceMacro(PAGEURL_KEY, bidRequest.params.pageurl, PAGEURL_DEFAULT, sspUrl);
-      sspUrl = _replaceMacro(CONTENTID_KEY, bidRequest.params.contentId, CONTENTID_DEFAULT, sspUrl);
-      sspUrl = _replaceMacro(CONTENTTITLE_KEY, bidRequest.params.contentTitle, CONTENTTITLE_DEFAULT, sspUrl);
-      sspUrl = _replaceMacro(CONTENTLENGTH_KEY, bidRequest.params.contentLength, CONTENTLENGTH_DEFAULT, sspUrl);
-      sspUrl = _replaceMacro(CONTENTURL_KEY, bidRequest.params.contentUrl, CONTENTURL_DEFAULT, sspUrl);
+      if (bidRequest.params.hasOwnProperty('output') && bidRequest.params.output != null) {
+        sspData.output = bidRequest.params.output;
+      } else {
+        sspData.output = PARAM_OUTPUT_DEFAULT;
+      }
+      if (bidRequest.params.hasOwnProperty('execution') && bidRequest.params.execution != null) {
+        sspData.execution = bidRequest.params.execution;
+      } else {
+        sspData.execution = PARAM_EXECUTION_DEFAULT;
+      }
+      if (bidRequest.params.hasOwnProperty('support') && bidRequest.params.support != null) {
+        sspData.support = bidRequest.params.support;
+      } else {
+        sspData.support = PARAM_SUPPORT_DEFAULT;
+      }
+      if (bidRequest.params.hasOwnProperty('playinit') && bidRequest.params.playinit != null) {
+        sspData.playinit = bidRequest.params.playinit;
+      } else {
+        sspData.playinit = PARAM_PLAYINIT_DEFAULT;
+      }
+      if (bidRequest.params.hasOwnProperty('volume') && bidRequest.params.volume != null) {
+        sspData.volume = bidRequest.params.volume;
+      } else {
+        sspData.volume = PARAM_VOLUME_DEFAULT;
+      }
+      if (playerWidth) {
+        sspData.width = playerWidth;
+      }
+      if (playerHeight) {
+        sspData.height = playerHeight;
+      }
+      if (bidRequest.params.hasOwnProperty('vpaidmode') && bidRequest.params.vpaidmode != null) {
+        sspData.vpaidmode = bidRequest.params.vpaidmode;
+      }
+      if (bidRequest.params.hasOwnProperty('appname') && bidRequest.params.appname != null) {
+        sspData.appname = bidRequest.params.appname;
+      }
+      if (bidRequest.params.hasOwnProperty('bundleid') && bidRequest.params.bundleid != null) {
+        sspData.bundleid = bidRequest.params.bundleid;
+      }
+      if (bidRequest.params.hasOwnProperty('aid') && bidRequest.params.aid != null) {
+        sspData.aid = bidRequest.params.aid;
+      }
+      if (bidRequest.params.hasOwnProperty('idfa') && bidRequest.params.idfa != null) {
+        sspData.idfa = bidRequest.params.idfa;
+      }
+      if (bidRequest.params.hasOwnProperty('gdpr') && bidRequest.params.gdpr != null) {
+        sspData.gdpr = bidRequest.params.gdpr;
+      }
+      if (bidRequest.params.hasOwnProperty('gdprcs') && bidRequest.params.gdprcs != null) {
+        sspData.gdprcs = bidRequest.params.gdprcs;
+      }
```

Comment:
just verifying that you expect `gdpr` params to be passed through in the bid params rather than the `bidderRequest.gdprConsent` object generally sent to `buildRequests()`: `buildRequests: function (bidRequests, bidderRequest) {`

Reply:
Thank you very much for confirming on this, it's an interesting point. You are correct that we are using custom parameters for our GDPR handling, even though there is likely overlap between your gdprConsent object and our gdprcs (consent string). I was unaware of bidderRequest.gdprConsent so I will have to familiarize myself with the documentation - we will likely add support through this property in a later pull request, but for the moment we are going to proceed with our own implementation. Thank you for bringing this feature to my attention!